Storage/History.swift

```diff
@@ -35,6 +35,7 @@ public protocol BrowserHistory {
     func setTopSitesCacheSize(size: Int32)
     func clearTopSitesCache() -> Success
     func refreshTopSitesCache() -> Success
+    func areTopSitesDirtyWithLimit(limit: Int) -> Deferred<Maybe<Bool>>
```

Comment:
nit: my gut is telling me that `func areTopSitesDirty(withLimit limit: Int) -> Deferred<Maybe<Bool>>` would be a slightly Swiftier name.